pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -331,3 +339,45 @@ func (d *dpiComponent) dpiAnnotations() map[string]string {
 	annotations[d.cfg.TyphaNodeTLS.NodeSecret.HashAnnotationKey()] = d.cfg.TyphaNodeTLS.NodeSecret.HashAnnotationValue()
 	return annotations
 }
+
+// This policy uses service selectors.
```

Comment:
Would like to see more detailed comments than this, especially at the function level. The fact that it uses service selectors should be obvious from reading the code, what's maybe less obvious is the purpose of this policy.

e.g., 

```
// dpiAllowTigeraPolicy creates a network policy which selects component X and allows the following traffic:
//
// ingress:
// - ...
//
// egress:
// - ...
//
// This policy is installed in Calico Enterprise management and managed clusters
```

Or something like that ^ 

This level of detail is probably something we should be writing in for each of these policies. Otherwise they start to just become "magic" and become difficult to maintain or modify over time.

Reply:
Agreed that this specific comment is not very useful.

After some thought I'm inclined to agree with your suggestion of having explicit comments to serve as documentation. I hesitate because I would have liked the code to self-document, and don't want comments to drift from implementation. But I personally can't see a way to improve the nature of how the code documents itself, and I agree that the way it documents itself is not sufficient due to how decentralized the self-documentation is